added trufflescripts functionality for master and adding addTokenPair…

W3stside opened the pull request:
Small changes/adds to truffle scripts for master and some features

can now add token pair without starting auction (was broken before)

W3stside committed:
added trufflescripts functionality for master and adding addTokenPair withoutstarting auction

Velenir commented on trufflescripts/give_tokens_and_deposit.js:
```diff
@@ -0,0 +1,52 @@
+  giveTokens(account, tokensToGive, master);
+
+  ({ ETH, GNO, TUL, OWL } = await getTokenBalances(account))
+  console.log(`Balance is:\t${ETH}\tETH,\t${GNO}\tGNO,\t${TUL}\tTUL,\t${OWL}\tOWL`)
```
But ultimately Balance doesn't change as what given is immediately deposited. What's the point to logging balances then?

W3stside replied:
just artifacts from copy pasting

Velenir commented on trufflescripts/add_token_pair.js:
```diff
@@ -58,44 +62,57 @@ module.exports = async () => {
+    // Deposit depends on ABOVE finishing first... so run here
+    await Promise.all(accounts.map(acct => Promise.all([
+      dx.deposit(sellToken.address, startingT1, { from: acct }),
+      // dx.deposit(buyToken.address, startingT2, { from: acct }),
```
why did you comment it out?

W3stside replied:
Proxy issue, fixing

Velenir commented on trufflescripts/add_token_pair.js:
```diff
@@ -1,7 +1,9 @@
 /* eslint no-console:0, no-multi-spaces:0, prefer-destructuring:1 */
+const { updateExchangeParams } = require('./utils/contracts')(artifacts)
```
If you are requiring `utils/contracts` may as well import `deployed`

W3stside replied:
ok

Velenir commented on trufflescripts/add_token_pair.js:
```diff
@@ -41,14 +43,16 @@ module.exports = async () => {
   const dx = await DutchExchange.at(Proxy.address)
   const eth = await TokenETH.deployed()
   const gno = await TokenGNO.deployed()
+  const owl = await TokenOWL.deployed()
```
I think, owl is also proxied, like dx, not sure if anything would break but...

W3stside replied:
yup

Velenir commented on trufflescripts/give_tokens_and_deposit.js:
```diff
@@ -0,0 +1,52 @@
+
+  const tokensToGive = { ETH: argv.eth, GNO: argv.gno, TUL: argv.tul, OWL: argv.owl }
+
+  giveTokens(account, tokensToGive, master);
```
`await giveTokens(...)`?

Velenir commented on trufflescripts/start_auction.js:
```diff
@@ -84,7 +85,7 @@ module.exports = async () => {
     [, account] = accounts
-  } else if (argv.seller) {
+  } else if (argv.seller || argv.master) {
     [account] = accounts
```
what's the use of `--master` if it doesn't change seller?

W3stside replied:
consistency, others use master flag

W3stside committed:
changed trufflescripts and fixed review requests
1. add_token_pair: fixed OWLProxy and uncommented depositing as was broken
2. get_auction_stats: made dynamic, can now get stats by pair specifying via --pair <st,bt>
3. sell_order: fixed review requests
4. contracts: added OWl in functionality
